nixos/modules/services/web-apps/trilium.nix

```diff
@@ -0,0 +1,102 @@
+{ config, lib, pkgs, ... }:
+
+let
+  cfg = config.services.trilium-server;
+in
+{
+
+  options.services.trilium-server = with lib; {
+    enable = mkEnableOption "trilium-server";
+
+    dataDir = mkOption {
+      type = types.str;
+      default = "/var/lib/trilium";
+      example = "/var/lib/trilium";
+      description = ''
+        The directory storing the nodes database and the configuration.
+      '';
+    };
+
+    instanceName = mkOption {
+      type = types.str;
+      default = "Trilium";
+      example = "Trilium";
+      description = ''
+        Instance name used to distinguish between different instances
+      '';
+    };
+
+    host = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      example = "127.0.0.1";
+      description = ''
+        The host address to bind to (defaults to localhost).
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 8080;
+      example = 8080;
+      description = ''
+        The port number to bind to.
+      '';
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    meta.maintainers = with lib.maintainers; [ kampka ];
+
+    users.groups.trilium = {};
+    users.users.trilium = {
+      description = "Trilium User";
+      group = "trilium";
+      home = cfg.dataDir;
+      isSystemUser = true;
+    };
+
+    systemd.services = {
+      trilium-server-setup = let 
+        configIni = pkgs.writeText "trilium-config.ini" ''
```

Comment:
Would you consider adding a `settings` option as described by https://github.com/NixOS/rfcs/pull/42?

Reply:
Looking at the RFC, there seems to be a lot of boilerplate still missing.
I am reading your comment as "Add a nested settings sub-option that can later be changed to use types.settings.ini / genINI without breaking compatibility." Did I get that right?

Reply:
That is the essence of my suggestion, yes.

I thought it would be nice to add some flexibility to the module. I didn't go through the full list of available options in upstream documentation, but I saw some options for serving over https which I assume would be of value to some people.

I've never used this application, or configured it, so if there really isn't anything of value that is a valid answer too.

Reply:
All remaining options concern https which I left out because it is so easy to just configure SSL offloading via nginx/caddy/etc. on NixOS that I didn't think anyone would bother, but I'm happy to add those if requested.

As for the flexibility argument, I agree that RFC 42 will give us that and I am absolutely in favor of adding it's implications to this module. I think however that a lot is still missing and compatibility depends highly on how the actual structure and semantics of the resulting types, especially how hierarchies are going to be represented. Since the configuration file in this case is INI format, there are probably close to a million opinions as to how that could go.

I'd propose filing a bug for this blocked RFC 42 that I'll happily tackle once that's formalized.

Reply:
If the only other options are related to web then I agree a reverse proxy seems like a good idea. Any interest in adding that option to this PR? Can always add at a later date if not.

:+1:

Reply:
Added a configuration option for nginx, I have no experience with other servers I'm afraid.